src/deploy/functions/validate.ts

```diff
@@ -35,9 +35,15 @@ export function functionsDirectoryExists(
  * @param functionNames Object containing function names as keys.
  * @throws { FirebaseError } Function names must be valid.
  */
-export function functionIdsAreValid(functions: { id: string }[]): void {
-  const validFunctionNameRegex = /^[a-zA-Z0-9_-]{1,63}$/;
-  const invalidIds = functions.filter((fn) => !validFunctionNameRegex.test(fn.id));
+export function functionIdsAreValid(functions: { id: string; platform: string }[]): void {
```

Comment:
Extraneous nit: There is now a `@types/cjson` and I actually already added it for a recent JS->TS refactor I did. Should be able to change to an import at the top.

Reply:
Actually, cjson isn't even in this file anymore. Yanked.